### PR TITLE
Social auth

### PR DESCRIPTION
**Pull request by supercede**
#### What does this PR do?
Integrate Facebook and Google login

#### Description of Task to be completed?
As a user, I can log in to Barefoot application via Facebook and Google.

#### How should this manually be Tested?
- `git clone` __https://github.com/devcareer/Dae-Alright-Backend__
- `cd` **Dae-Alright-Backend**
- `git checkout <branch-name>`
- `npm install`
- `npm test`

#### What are the relevant ~~Pivotal Tracker~~ **Trello** stories?
Currently, a new user cannot log in with their pre-existing social media accounts. He/She can only login with email password combinations.

As a user, I should be able to log in via Google and Facebook

### CODE REVIEW

**Comment by supercede** [comment]
[As a user, I can log in to Barefoot application via Facebook and Google.](https://trello.com/c/Gro698KG/4-as-a-user-i-can-log-in-to-barefoot-application-via-facebook-and-google)

**Comment by supercede** [comment]
[As a user, I can log in to Barefoot application via Facebook and Google.](https://trello.com/c/Gro698KG/4-as-a-user-i-can-log-in-to-barefoot-application-via-facebook-and-google)

**Review by TundeMercy — COMMENTED** [review]
Please run `npm run lint` and correct all formating errors that will be shown

**Comment by TundeMercy** [comment]
Please, can you squash all your six commits into one?

File: src/controllers/auth.controller.js [hunk 1]
```diff
@@ -2,54 +2,68 @@ import database from '../database/models';
 import { generateToken } from '../helpers/auth';
 import { errorResponse, successResponse, serverError } from '../helpers/response';
 
-const { User } = database;
+const { Vendor, User } = database;
 
 /**
- * Creates a user successfully
+ * Creates a client successfully
```
Comment (TundeMercy) [inline src/controllers/auth.controller.js:8]: How are these changes attributed to your commit?

File: src/controllers/auth.controller.js [hunk 2]
```diff
@@ -2,54 +2,68 @@ import database from '../database/models';
...
  * @returns {json} The response from db or error.
  */
-export const createUser = async (req, res) => {
+export const createClient = async (req, res) => {
+  const role = req.url.split('/')[1];
   try {
-    const user = await User.create(req.body);
-    const newUser = await user.getSafeDataValues();
-    const token = generateToken(newUser);
+    const client = role === 'vendor' ? await Vendor.create(req.body) : await User.create(req.body);
```
Comment (TundeMercy) [inline src/controllers/auth.controller.js:17]: All these ??

File: src/database/models/user.js [hunk 3]
```diff
@@ -40,8 +45,10 @@ export default (sequelize, DataTypes) => {
   });
 
   User.prototype.generatePasswordHash = async function generatePasswordHash() {
-    const saltRounds = +process.env.SALT;
-    return bcrypt.hash(this.password, saltRounds);
+    if (!this.provider) {
+      const saltRounds = +process.env.SALT;
```
Comment (TundeMercy) [inline src/database/models/user.js:49]: You may need to revert your changes in all these files. These are all done and contributed by a different person

**Review by meetKazuki — APPROVED** [review]
May the Force be with us. Nice work👍

**Pull request merged**